src/model/liveselection.js

```diff
@@ -565,6 +567,11 @@ export default class LiveSelection extends Selection {
 
 			// ...look for a first character node in that range and take attributes from it.
 			for ( const item of range ) {
+				// If the item is an object, we don't want to get attributes from its children.
+				if ( item.item instanceof Element && schema.objects.has( item.item.name ) ) {
+					break;
```

Comment:
After adding this test - https://github.com/ckeditor/ckeditor5-engine/pull/990/commits/3fddb2a8f28af3c21a4944688feb043a696f3785#diff-2d9aba768c395558c529a50282531971R898 - the test will fail if we use `return` instead of `break`.